noir-projects/aztec-nr/aztec/src/encrypted_logs/payload.nr

```diff
@@ -94,23 +90,47 @@ pub fn compute_partial_public_log_payload<let P: u32, let M: u32>(
     sender: AztecAddress,
     plaintext: [u8; P],
 ) -> [u8; M] {
-    let extended_plaintext: [u8; P + 1] = extend_private_log_plaintext(plaintext);
-    compute_encrypted_log(
+    let extended_plaintext: [u8; P + PLAINTEXT_LENGTH_SIZE] =
+        extend_private_log_plaintext(plaintext);
+    let encrypted: [u8; M - 32] = compute_encrypted_log(
         contract_address,
         ovsk_app,
         ovpk,
         recipient,
-        sender,
         extended_plaintext,
-    )
+    );
+
+    // We assume that the sender wants for the recipient to find the tagged note, and therefore that they will cooperate
+    // and use the correct tag. Usage of a bad tag will result in the recipient not being able to find the note
+    // automatically.
+    let tag = unsafe { get_app_tag_as_sender(sender, recipient) };
+    increment_app_tagging_secret_index_as_sender(sender, recipient);
+    // Silo the tag with contract address.
+    // This is done by the kernel circuit to the private logs.
+    // But since the partial log will be finalized and emitted in public as unencrypted log, its tag is not siloed at the moment.
+    // To avoid querying logs using two types of tags, we silo the tag manually here.
+    // This should be done by the AVM when its processing the raw logs instead of their hashes.
+    let siloed_tag_bytes: [u8; 32] =
```

Comment:
```suggestion
    // Silo the tag with contract address.
    // This is done by the kernel circuit to the private logs, but since the partial log will be finalized and emitted in
    // public as unencrypted log, its tag is not siloed at the moment.
    // To avoid querying logs using two types of tags, we silo the tag manually here.
    // TODO(#10273) This should be done by the AVM when its processing the raw logs instead of their hashes.
    let siloed_tag_bytes: [u8; 32] =
```

Reply:
Created https://github.com/AztecProtocol/aztec-packages/issues/10273 to track this